core/chains/evm/txmgr/stuck_tx_detector.go

```diff
@@ -217,16 +217,29 @@ func (d *stuckTxDetector) detectStuckTransactionsHeuristic(ctx context.Context,
 		}
 		// Tx attempts are loaded from newest to oldest
 		oldestBroadcastAttempt, newestBroadcastAttempt, broadcastedAttemptsCount := findBroadcastedAttempts(tx)
+		d.lggr.Debugf("found %d broadcasted attempts for tx id %d in stuck transaction heuristic", broadcastedAttemptsCount, tx.ID)
+
 		// 2. Check if Threshold amount of blocks have passed since the oldest attempt's broadcast block num
+		if oldestBroadcastAttempt == nil {
+			d.lggr.Debugw("failed to find broadcast attempt for tx in stuck transaction heuristic", "tx", tx)
+			continue
+		}
+
+		if oldestBroadcastAttempt.BroadcastBeforeBlockNum == nil {
+			d.lggr.Debugw("BroadcastBeforeBlockNum was not set for broadcast attempt in stuck transaction heuristic", "attempt", oldestBroadcastAttempt)
+			continue
+		}
+
 		if *oldestBroadcastAttempt.BroadcastBeforeBlockNum > blockNum-int64(*d.cfg.Threshold()) {
 			continue
 		}
+
 		// 3. Check if the transaction has at least MinAttempts amount of broadcasted attempts
 		if broadcastedAttemptsCount < *d.cfg.MinAttempts() {
 			continue
 		}
 		// 4. Check if the newest broadcasted attempt's gas price is higher than what our gas estimator's GetFee method returns
-		if compareGasFees(newestBroadcastAttempt.TxFee, marketGasPrice) <= 0 {
+		if newestBroadcastAttempt != nil && compareGasFees(newestBroadcastAttempt.TxFee, marketGasPrice) <= 0 {
```

Comment:
if this is nil, shouldn't we log an error, similar to if `oldestBroadcastAttempt` is nil?

Would it make sense to move both checks up to line 220 before we do anything else? Although interestingly it looks like if oldest is nil we continue and if newest is nil we will add it to the stuck txns.